registry/remote/repository.go

```diff
@@ -548,6 +548,11 @@ func (r *Repository) referrersPageByAPI(ctx context.Context, artifactType string
 		return "", errutil.ParseErrorResponse(resp)
 	}
 
+	if resp.Header.Get("Content-Type") != ocispec.MediaTypeImageIndex {
+		resp.StatusCode = http.StatusNotFound
+		return "", errutil.ParseErrorResponse(resp)
```

Comment:
We should modify the response status code. Instead, we can do something like
```go
err := errutil.ParseErrorResponse(resp)
return "", fmt.Errorf("unknown content returned, expecting image index: %w", err)
```
You may improve the error message :P

Reply:
I am using `errors.Join` to return multiple errors from the function.